### PR TITLE
Service tag support

### PR DESCRIPTION
**Pull request by ashutosh-mishra**

### CODE REVIEW

**Comment by ashutosh-mishra** [comment]
@njagadish @hemanthravi @YogeshRajmane

File: gbpservice/nfp/orchestrator/modules/service_orchestrator.py [hunk 1]
```diff
@@ -434,22 +436,17 @@ def create_network_function(self, context, network_function_info):
...
-
-            '''#(TODO) make changes for config_init and ansible too
-            template = service_config.split('heat_config:')[1]
-            network_function = {
-                'service_config': template
-            }
-            network_function = self.db_handler.update_network_function(
-                self.db_session, network_function_id, network_function)
-            '''
+            if tag_str != 'config_init:':
```
Comment (hemanthravi) [inline gbpservice/nfp/orchestrator/modules/service_orchestrator.py]: What does base_mode_support indicate? This parsing applies to both base_mode and advance_mode.

File: gbpservice/nfp/orchestrator/config_drivers/heat_driver.py [hunk 2]
```diff
@@ -644,10 +644,11 @@ def _update_node_config(self, auth_token, tenant_id, service_profile,
         base_mode_support = (True if service_details['device_type'] == 'None'
                              else False)
 
-        stack_template = service_chain_node.get('config')
-        stack_template = (jsonutils.loads(stack_template) if
-                          stack_template.startswith('{') else
-                          yaml.load(stack_template))
+        _, stack_template_str = self.parse_template_config_string(
```
Comment (hemanthravi) [inline gbpservice/nfp/orchestrator/config_drivers/heat_driver.py:647]: Can tag_str be any of the other values such as 'config_init' and 'ansible' here?

File: gbpservice/nfp/orchestrator/config_drivers/heat_driver.py [hunk 3]
```diff
@@ -787,6 +788,20 @@ def _update_node_config(self, auth_token, tenant_id, service_profile,
...
 
+    def parse_template_config_string(self, config_str):
+        service_config = tag_str = ''
+        for tag_str in [nfp_constants.HEAT_CONFIG_TAG,
+                        nfp_constants.CONFIG_INIT_TAG,
+                        nfp_constants.ANSIBLE_TAG]:
+            try:
+                service_config = config_str.split(tag_str + ':')[1]
+                break
+            except Exception:
```
Comment (njagadish) [inline gbpservice/nfp/orchestrator/config_drivers/heat_driver.py]: We should not use exception to proceed in positive cases. What if you get a different exception in which case we should not proceed. Rather check exact case and decide when to proceed and when not to.

File: gbpservice/nfp/orchestrator/config_drivers/heat_driver.py (hunk 2 above)
Comment (njagadish) [inline gbpservice/nfp/orchestrator/config_drivers/heat_driver.py:647]: Talked to Ashutosh. This call will get heat_config always as SO filters it.

**Comment by njagadish** [comment]
Changes look fine.

**Pull request merged**